hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -559,18 +569,22 @@ class Resolver(tl: TraceLogger)
         case S(defn @ CallableDefinition(params = ps :: pss)) =>
           val (argCountUB, argCountLB) = as match
           // Tup: regular arguments
-          case tup: Term.Tup => (
-            !tup.fields.exists(_.isInstanceOf[Spd]),
-            tup.fields.map:
-              case Fld(asc = S(_)) => 0
-              case _: Fld => 1
-              case _: Spd => 0
-            .sum +
-            tup.fields.exists:
-              case Fld(asc = S(_)) => true
-              case _ => false
-            .into(if _ then 1 else 0)
-          )
+          case tup: (Term.Tup | Term.CtxTup) => 
+            val fields = tup match
+              case Term.Tup(fs) => fs
+              case Term.CtxTup(fs) => fs
+            (
+              !fields.exists(_.isInstanceOf[Spd]),
+              fields.map:
+                case Fld(asc = S(_)) => 0
+                case _: Fld => 1
+                case _: Spd => 0
+              .sum +
+              fields.exists:
+                case Fld(asc = S(_)) => true
+                case _ => false
+              .into(if _ then 1 else 0)
```

Comment:
Use `.count`?

Reply:
This is different from the above case: all fields with ascriptions are aggregated to one record and passed to the function